src/events/backup.py

```diff
@@ -92,7 +92,6 @@ def _on_s3_credentials_gone(self, event: CredentialsGoneEvent):
         self.charm.state.cluster.update({"s3-credentials": ""})
 
     def _on_create_backup_action(self, event: ActionEvent):
-        # TODO
         failure_conditions = [
             (not self.charm.unit.is_leader(), "Action must be ran on the application leader"),
```

Comment:
**question**: We do not need this at all, should we keep it? Is it just "a good practice to enforce" to run actions on the leader?

Reply:
I think limiting it to the leader is a safe bet. The leader is always a single unit that effectively blocks execution to a single action process at a time, it acts as a semaphore to avoid running backups concurrently on multiple units.
Let's keep it.